docs/http.rst

```diff
@@ -0,0 +1,479 @@
+as "secret" in the registration response.
+
+Lexicon
+-------
+
+   :{UAID}: The Push User Agent Registration ID
+
+Push assigns each remote recipient a unique identifier. This value is
+assigned during **Registration**
+
+   :{CHID}: The :term:`Channel` Subscription ID
+
+Push assigns a unique identifier for each subscription for a given {UAID}.
+This value is assigned during **Channel Subscription**
+
+   :{message-id}: The unique Message ID
+
+Push assigns each message for a given Channel Subscription a unique
+identifier. This value is assigned during **Send Notification**
+
+Response
+--------
+
+The responses will be JSON formatted objects. In addition, API calls
+will return valid HTTP error codes (see :ref:`errors` sub-section for
+descriptions of specific errors).
+
+For non-success responses, an extended error code object will be
+returned with the following format:
+
+.. code-block:: json
+
+    {
+        "code": 404,  // matches the HTTP status code
+        "errno": 103, // stable application-level error number
+        "error": "Not Found", // string representation of the status
+        "message": "No message found" // optional additional error information
+    }
+
+
+.. _errors:
+
+Error Codes
+-----------
+
+Unless otherwise specified, all calls return the following error codes:
+
+-  20x - Success
+-  301 - Moved + `Location:` if `{token}` is invalid (Bridge API Only)
+-  400 - Bad Parameters
+
+   - errno 101 - Missing neccessary crypto keys
+   - errno 108 - Router type is invalid
+   - errno 110 - Invalid crypto keys specified
+
+-  401 - Bad Authorization
+
+   - errno 109 - Invalid authentication
+
+-  404 - `{UAID}` not found or invalid
+
+   - errno 102 - Invalid URL endpoint
+   - errno 103 - Expired URL endpoint
+   - errno 105 - Endpoint became unavailable during request
+   - errno 106 - Invalid subscription
+
+-  500 - Unknown server error
+
+   - errno 999 - Unknown error
+
+-  503 - Server temporarily unavaliable.
+
+   -  errno 201 - Use exponential back-off for retries
+   -  errno 202 - Immediate retry ok
+
+Calls
+-----
+
+.. _send:
+
+Send Notification
+~~~~~~~~~~~~~~~~~
+
+Send a notification to the given endpoint identified by it's `token`.
+
+Call:
+^^^^^
+.. http:put:: /push/{token}
+
+    If the client is using webpush style data delivery, then the body in its
+    entirety will be regarded as the data payload for the message per
+    `the WebPush spec
+    <https://tools.ietf.org/html/draft-thomson-webpush-http2-02#section-5>`_.
+
+Parameters:
+^^^^^^^^^^^
+
+    :form version: (*Optional*) Version of notification, defaults to current
+                   time
+
+Reply:
+^^^^^^
+
+.. code-block:: json
+
+    {"message-id": {message-id}}
+
+Errors:
+^^^^^^^
+    :statuscode 404: `token` is invalid.
+    :statuscode 202: Message stored for delivery to client at a later
+                     time.
+    :statuscode 200: Message delivered to node client is connected to.
```

Comment:
Is it possible to remove some of these header sections and still maintain readability? They add a lot of padding and it seems like it should be obvious that the status codes are in the reply, and the call portion should be formatted appropriately so that its obvious that is the HTTP call being done.